### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by paulstapor**
Some changes, bugfixes and enhancements from the develop-branch should be ready to merge now.

### CODE REVIEW

File: @PestoOptions/PestoOptions.m [hunk 1]
```diff
@@ -20,6 +20,15 @@
 
         obj_type = 'log-posterior';
         
+        % Maximum number of putputs, the objective function can provide:
```
Comment (FFroehlich) [inline @PestoOptions/PestoOptions.m]: putputs -> outputs

File: private/objectiveWrap.m [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    case 3
+                        [J, G, H] = objectiveFunction(theta);
+                end
+
+                switch type
+                    case 'log-posterior'          , varargout = {-J,-G(I),-H(I,I)};
+                    case 'negative log-posterior' , varargout = { J, G(I), H(I,I)};
+                end
+                if(any(isnan(H)))
+                    error('Hessian contains NaNs')
```
Comment (FFroehlich) [inline private/objectiveWrap.m]: me gusta

File: private/objectiveWrapWErrorCount.m [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    case 'log-posterior'          , varargout = {-J,-G(I),-H(I,I)};
+                    case 'negative log-posterior' , varargout = { J, G(I), H(I,I)};
+                end
+                if(any(isnan(H)))
+                    error('Hessian contains NaNs')
+                end
+        end
+
+        % Reset error count
+        error_count = 0;
```
Comment (FFroehlich) [inline private/objectiveWrapWErrorCount.m]: please do not reset error count but reduce by 1

File: private/objectiveWrap.m (hunk 2 above)
Comment (paulstapor) [inline private/objectiveWrap.m]: do you mean this shouldn't give an error? Or... ?

Comment (paulstapor) [inline private/objectiveWrap.m]: Fixed those things

Comment (FFroehlich) [inline private/objectiveWrap.m]: no I like that we have a check now (if there are NaNs in H but not in J and G, fmincon usually goes into infinite loops). but I just noticed that H is a matrix so you should probably check if(any(isnan(H(:))))

Comment (paulstapor) [inline private/objectiveWrap.m]: fixed that, too. Ready to merge?

**Pull request merged**